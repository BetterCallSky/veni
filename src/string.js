import _makeError from './_makeError';

export default (msg = 'must be a string') => (data) => {
  if (data.value == null) {
    return {
      stop: true,
    };
  }
  if (typeof data.value !== 'string') {
    return {
      stop: true,
      error: _makeError(data, msg),
    };
  }
  return undefined;
};
