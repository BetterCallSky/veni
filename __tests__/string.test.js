import string from '../src/string';

describe('string', () => {
  [undefined, null].forEach(value => {
    it(`should ignore optional: ${value}`, () => {
      expect(string()({value})).toMatchSnapshot();
    });
  });

  [1, {}, [1, 2, 3]].forEach(value => {
    it(`should return an error for: ${value}`, () => {
      expect(string()({value})).toMatchSnapshot();
    });
  });

  it('should return a custom message', () => {
    expect(string('custom message')({value: 12})).toMatchSnapshot();
  });

  it('should return a value for a valid string', () => {
    expect(string()({value: 'abc'})).toMatchSnapshot();
  });
});
