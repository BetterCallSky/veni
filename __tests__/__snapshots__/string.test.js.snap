// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string should ignore optional: null 1`] = `
Object {
  "errors": Array [],
  "value": null,
}
`;

exports[`string should ignore optional: undefined 1`] = `
Object {
  "errors": Array [],
  "value": undefined,
}
`;

exports[`string should return a custom message 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "custom message",
      "path": "value",
    },
  ],
  "value": 12,
}
`;

exports[`string should return a value for a valid string 1`] = `
Object {
  "errors": Array [],
  "value": "abc",
}
`;

exports[`string should return an error for: [object Object] 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "must be a string",
      "path": "value",
    },
  ],
  "value": Object {},
}
`;

exports[`string should return an error for: 1 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "must be a string",
      "path": "value",
    },
  ],
  "value": 1,
}
`;

exports[`string should return an error for: 1,2,3 1`] = `
Object {
  "errors": Array [
    Object {
      "error": "must be a string",
      "path": "value",
    },
  ],
  "value": Array [
    1,
    2,
    3,
  ],
}
`;
